File: src/main/java/FizzBuzz.java
public class FizzBuzz {

	private static final int NUM_FOR_FIZZ = 3;
	private static final int NUM_FOR_BUZZ = 5;

	private static final String FIZZ = "Fizz";
	private static final String BUZZ = "Buzz";
	private static final String FIZZ_BUZZ = FIZZ + " " + BUZZ;

	public static void main(String[] args) {
		for (int i = 1; i <= 100; i++) {
			if (isFizz(i) && isBuzz(i)) {
				System.out.println(FIZZ_BUZZ);
			} else if (isFizz(i)) {
				System.out.println(FIZZ);
			} else if (isBuzz(i)) {
				System.out.println(BUZZ);
			} else {
				System.out.println(i);
			}
		}
	}

	private static boolean isFizz(int i) {
		return i % NUM_FOR_FIZZ == 0;
	}

	private static boolean isBuzz(int i) {
		return i % NUM_FOR_BUZZ == 0;
	}
}

File: src/main/kotlin/FizzBuzz.kt
private const val NUM_FOR_FIZZ = 3
private const val NUM_FOR_BUZZ = 5

private const val FIZZ = "Fizz"
private const val BUZZ = "Buzz"
private const val FIZZ_BUZZ = "$FIZZ $BUZZ"

fun main(arg: Array<String>){
    for (i in 1..100){
        when{
            isBuzz(i) && isFizz(i) -> println(FIZZ_BUZZ)
            isFizz(i) -> println(FIZZ)
            isBuzz(i) -> println(BUZZ)
            else -> println(i)
        }
    }
}

private fun isFizz(i: Int): Boolean {
    return i % NUM_FOR_FIZZ == 0
}

private fun isBuzz(i: Int): Boolean {
    return i % NUM_FOR_BUZZ == 0
}